kmitl_purchase_request_2_new/models/purchase_request_two.py

```diff
@@ -0,0 +1,179 @@
+# -*- coding: utf-8 -*-
+import logging
+
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError, ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class PurchaseRequestTwo(models.Model):
+    _name = 'purchase.request.two'
+    _description = 'PurchaseRequestTwo'
+
+    name = fields.Char(
+        string='Reference', 
+        required=True, 
+        copy=False, 
+        readonly=True, 
+        default='New')
+
+    pr1_ref = fields.Many2one(
+        'purchase.request',
+        string='อ้างอิง PR1',
+        readonly=True
+    )
+
+    vendor = fields.Many2one(
+        "res.partner",
+        string="Vendor",
+        help="Select a vendor to create a purchase order for the selected request lines.",
+    )
+    start_date = fields.Date(
+        string="วันที่เริ่มสัญญา",
+        help="The start date for the purchase order. If not set, the current date will be used.",
+    )
+    end_date = fields.Date(
+        string="วันที่สิ้นสุดสัญญา",
+        help="The end date for the purchase order. If not set, the start date will be used.",
+    )
+    purchase_request_number = fields.Char(
+        string="หมายเลขคำสั่งซื้อ",
+        required=True,
+        help="The number of the purchase request associated with the selected lines.",
+    )
+    purchase_type = fields.Selection(
+        [("order", "ใบสั่งซื้อ/จ้าง"), ("procurement", "สัญญาซื้อข้าย"), ("construction", "สัญญาจ้างก่อสร้าง")],
+        string="ประเภทสัญญา",
+        help="Select the type of purchase order to create. Standard for regular orders, Urgent for expedited orders.",
+    )
+    payment_type = fields.Selection([
+        ("direct", "จ่ายตรง"),
+        ("loan", "เงินยืม"),
+        ("prepaid", "สำรองจ่าย")
+    ], string="ประเภทการจ่ายเงิน", readonly=True)
+
+    purchase_request_name = fields.Char(
+        string="ชื่อใบสั่งซื้อ/จ้าง",
+        help="The name of the purchase request associated with the selected lines.",
+    )
+
+    line_ids = fields.One2many('purchase.request.two.line', 'pr2_id', string='Products')
+
+    state = fields.Selection([
+        ('draft', 'Draft'),
+        ('submitted', 'Submitted'),
+        ('approved', 'Approved'),
+        ('po_created', 'PO Created'),
+        ('rejected', 'Rejected')
+    ], default='draft', string='Status', tracking=True)
```

Comment:
```suggestion
    state = fields.Selection(_STATES, default='draft', string='Status', tracking=True)
```
แนะนำให้สร้างเป็นตัวแปรดีกว่า